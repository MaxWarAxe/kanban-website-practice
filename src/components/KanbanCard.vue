<script setup lang="ts">
    import { computed } from 'vue';
    import { Performer } from '@/classes/Performer'

    const props = defineProps<{
        title: string,
        content: string,
        performer: Performer | null
    }>();    
    let nameFirstLetter = computed (() => {
        return props.performer.name[0]
    })
    let isPerformerExists = computed (() => {
        if (props.performer == null)
            return false
        else
            return true
    })
</script>

<template>
    <div class="card-container">
        <header class="card-container__header">
            {{ props.title }}
        </header>
        <main class="card-container__content">
            {{ props.content }}
        </main>
        <footer class="card-container__utility">
            <div class="card-container__profile">
                <div class="card-container__picture"
                     data-test="performer-icon"
                     >
                    <div v-if = "!isPerformerExists">
                        ?
                    </div>
                    <div v-else>
                        {{ nameFirstLetter }}
                    </div>
                </div>
                <div v-if="isPerformerExists" 
                     class="card-container__performer-name"
                     data-test="performer-name"
                     >
                     {{ props.performer.name }}
                </div>
            </div>
        </footer>
    </div>
</template>

<style lang="scss" scoped>
    .card-container{
        width: 330px;
        height: auto;
        display: flex;
        flex-direction: column;
        gap: 10px;
        background-color: var(--primary-bg-color);
        border-radius: 10px;
        box-shadow: 0px 5px 0px var(--primary-border-color);
        padding: 10px;
        transition: 0.2s;
        &__header{
            font-size: 18px;
            font-weight: 600;
            color: var(--primary-text-color);
            overflow: hidden;
        }
        &__content{
            font-size: 14px;
            font-weight: 400;
            color: var(--primary-text-color);
            overflow: hidden;
        }
        &__utility{
            height: 40px;
            width: 100%;
        }
        &__profile{
            display: flex;
            align-items: center;
            gap: 5px;
        }
        &__picture{
            width: 40px;
            height: 40px;
            display: flex;
            justify-content: center;
            align-items: center;
            font-size: 20px;
            color: var(--button-text-color);
            border-radius: 50%;
            background-color: var(--primary-icon-color);
        }
        &__performer-name{
            font-size: 12px;
            color: var(--primary-text-color);
        }
        &:hover{
            background-color: var(--primary-selection-color);
            cursor: grab;
        }
    }
</style>