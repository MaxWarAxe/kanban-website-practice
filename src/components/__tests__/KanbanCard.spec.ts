

import { describe,it,expect } from 'vitest'
import { mount } from '@vue/test-utils'
import KanbanCard from '@/components/KanbanCard.vue'


describe('KanbanCard', () => {

    it('Коректно рендерится', ()=>{
        const wrapper = mount(KanbanCard)
        expect(wrapper.html()).toMatchSnapshot()
    })

    it('Пропсы должны корректно рендериться', async() => {
        const wrapper = mount(KanbanCard,{
            props : {
                title: 'Тестовое название',
                content: 'Тестовый контент',
                performer: {id : 1, name: 'Тестовый исполнитель'}
            }
        })
        expect(wrapper.find('header').text()).toEqual('Тестовое название')
        expect(wrapper.find('main').text()).toEqual('Тестовый контент')
        expect(wrapper.get('[data-test="performer-name"]').text()).toEqual('Тестовый исполнитель')
    })

    it('Иконка исполнителя корректно рендерится', async() => {
        const wrapper = mount(KanbanCard,{
            props : {
                title: 'Тестовое название',
                content: 'Тестовый контент',
                performer: {id : 1, name: 'Тестовый исполнитель'}
            }
        })
        expect(wrapper.get('[data-test="performer-icon"]').text()).toEqual('Т')
        const wrapper2 = mount(KanbanCard,{
            props : {
                title: 'Тестовое название',
                content: 'Тестовый контент',
                performer: null
            }
        })
        expect(wrapper2.get('[data-test="performer-icon"]').text()).toEqual('?')
    })

})