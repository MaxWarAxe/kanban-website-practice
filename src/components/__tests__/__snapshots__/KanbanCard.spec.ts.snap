// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`KanbanCard > Коректно рендерится 1`] = `
"<div data-v-45fb8c1e="" class="card-container">
  <header data-v-45fb8c1e="" class="card-container__header"></header>
  <main data-v-45fb8c1e="" class="card-container__content"></main>
  <footer data-v-45fb8c1e="" class="card-container__utility">
    <div data-v-45fb8c1e="" class="card-container__profile">
      <div data-v-45fb8c1e="" class="card-container__picture" data-test="performer-icon">
        <div data-v-45fb8c1e=""> ? </div>
      </div>
      <!--v-if-->
    </div>
  </footer>
</div>"
`;
